pkg/spiffecert/task_queue.go

```diff
@@ -0,0 +1,106 @@
+// Package spiffecert contains code related to spiffe identity management
+package spiffecert
+
+import (
+	"log"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+)
+
+// TaskQueue manages a work queue through an independent worker that
+// invokes the given sync function for every work item inserted.
+type TaskQueue struct {
```

Comment:
Because this queue is generic enough and not tied directly to spire events, we could have it live in its own package